src/cli/lint/internalLints/upgrade/patches/8.2.1/collapseDuplicateMergeKeys.ts

```diff
@@ -0,0 +1,81 @@
+import { inspect } from 'util';
+
+import { glob } from 'fast-glob';
+import { promises as fs } from 'fs-extra';
+
+import type { PatchFunction, PatchReturnType } from '../..';
+import { log } from '../../../../../../utils/logging';
+
+const collapseDuplicateMergeKeys: PatchFunction = async ({
+  mode,
+}): Promise<PatchReturnType> => {
+  const buildkiteFiles = await glob(
+    ['.buildkite/**/*.yml', '.buildkite/**/*.yaml'],
+    { onlyFiles: true },
+  );
+
+  if (buildkiteFiles.length === 0) {
+    return { result: 'skip', reason: 'no Buildkite files found' };
+  }
+
+  const input = await Promise.all(
+    buildkiteFiles.map((name) => fs.readFile(name, 'utf-8')),
+  );
+
+  const replaced = await Promise.all(
+    input.map(collapseDuplicateMergeKeysInFile),
+  );
+
+  if (replaced.every((r, i) => r === input[i])) {
+    return { result: 'skip', reason: 'no duplicate merge keys found' };
+  }
+
+  if (mode === 'lint') {
+    return { result: 'apply' };
+  }
+
+  await Promise.all(
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    buildkiteFiles.map((name, i) => fs.writeFile(name, replaced[i]!)),
```

Comment:
nit: I don't imagine this will be too many files but do we want to avoid writing where it hasn't changed?